### PR TITLE
Incoming Call and In-Call new UI initial implementation

### PR DESCRIPTION
**Pull request by Semrubo**

### CODE REVIEW

File: ace-android.iml [hunk 1]
```diff
@@ -92,8 +92,8 @@
     <orderEntry type="sourceFolder" forTests="false" />
     <orderEntry type="library" exported="" name="gcm" level="project" />
     <orderEntry type="library" exported="" name="org.xbill.dns_2.1.7" level="project" />
-    <orderEntry type="library" exported="" name="maven-ant-tasks-2.1.3" level="project" />
     <orderEntry type="library" exported="" name="huelocalsdk" level="project" />
+    <orderEntry type="library" exported="" name="maven-ant-tasks-2.1.3" level="project" />
```
Comment (ianblenke) [inline ace-android.iml]: Did changing the order of this matter for some reason?

File: src/org/linphone/IncomingCallActivity.java [hunk 2]
```diff
@@ -207,16 +207,27 @@ protected void onResume() {
...
 		} else {
-			mNumberView.setText(address.asStringUriOnly());
+			mNameView.setText(contact.getName());
 		}
+		// To be done after findUriPictureOfContactAndSetDisplayName called
+		//TODO: question????lame_if_unknown)) {
+//			mNumberView.setText(address.getUserName());
+//		} else {
+//			mNumberView.setText(address.asStringUriOnly());
+//		}
```
Comment (ianblenke) [inline src/org/linphone/IncomingCallActivity.java]: It looks like the above commented-out lines are actually included above. Can we cut these out?

File: res/layout/item_in_call_second_flashing.xml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/layout_item_in_call_incoming"
+
+    android:orientation="horizontal"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="40dp"
+    android:padding="5dp"
+    android:background="@color/incomming_header_drak_bg_transparent"
```
Comment (ianblenke) [inline res/layout/item_in_call_second_flashing.xml]: Please change this from "incomming" to "incoming".

File: res/layout/item_in_call_second_passive.xml [hunk 4]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/layout_item_in_call_passive"
+    android:orientation="horizontal"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="40dp"
+    android:padding="5dp"
+    android:background="@color/incomming_header_drak_bg_transparent"
```
Comment (ianblenke) [inline res/layout/item_in_call_second_passive.xml]: Please change this from "incomming" to "incoming".

File: res/values/color.xml [hunk 5]
```diff
@@ -10,6 +11,13 @@
     <color name="text_default">#ff5b656f</color>
     <color name="text_disabled">#ffa6b3bf</color>-->
 
+    <color name="incomming_btn_drak_backgtound">#4E4242</color>
+    <color name="incomming_light_backgtound">#FF7100</color>
+
+
+    <color name="incomming_header_drak_bg_transparent">#CC222426</color>
+    <color name="incomming_header_drak_bg">#222426</color>
```
Comment (ianblenke) [inline res/values/color.xml]: Please change these from "incomming" to "incoming".

File: src/org/linphone/IncomingCallActivity.java [hunk 6]
```diff
@@ -247,19 +258,17 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
...
+		flashOrangeBackgroundTimer.schedule(new TimerTask() {
 			@Override
 			public void run() {
 				IncomingCallActivity.this.runOnUiThread(new Runnable() {
 					@Override
 					public void run() {
-						Integer colorFrom = Color.TRANSPARENT;
-						Integer colorTo = Color.rgb(90, 17, 17);//RED;
+						Integer colorFrom = getResources().getColor(R.color.incomming_header_drak_bg);
+						Integer colorTo = getResources().getColor(R.color.incomming_light_backgtound);
```
Comment (ianblenke) [inline src/org/linphone/IncomingCallActivity.java]: Please change these from "incomming" to "incoming".

File: src/org/linphone/LinphoneActivity.java [hunk 7]
```diff
@@ -338,10 +338,12 @@ public void registrationState(LinphoneCore lc, LinphoneProxyConfig proxy, Linpho
...
 			@Override
 			public void callState(LinphoneCore lc, LinphoneCall call, LinphoneCall.State state, String message) {
-
+				//InCallActivity will handle it
+				if (InCallActivity.isInstanciated())
+					return;
 				if (state == State.IncomingReceived) {
-
-					startActivity(new Intent(LinphoneActivity.instance(), IncomingCallActivity.class));
+					LinphoneManager.startIncommingCallActivity(LinphoneActivity.this);
```
Comment (ianblenke) [inline src/org/linphone/LinphoneActivity.java]: Please change this from "Incomming" to "Incoming".

File: src/org/linphone/LinphoneActivity.java [hunk 8]
```diff
@@ -1264,7 +1266,8 @@ public void resetClassicMenuLayoutAndGoBackToCallIfStillRunning() {
 		if (LinphoneManager.isInstanciated() && getLc().getCallsNb() > 0) {
 			LinphoneCall call = getLc().getCalls()[0];
 			if (call.getState() == LinphoneCall.State.IncomingReceived) {
-				startActivity(new Intent(LinphoneActivity.this, IncomingCallActivity.class));
+				LinphoneManager.startIncommingCallActivity(this);
```
Comment (ianblenke) [inline src/org/linphone/LinphoneActivity.java]: Please change this from "Incomming" to "Incoming".

File: src/org/linphone/LinphoneActivity.java [hunk 9]
```diff
@@ -1414,7 +1417,8 @@ protected void onResume() {
 				LinphoneCall call = getLc().getCalls()[0];
 				LinphoneCall.State callState = call.getState();
 				if (callState == State.IncomingReceived) {
-					startActivity(new Intent(this, IncomingCallActivity.class));
+					LinphoneManager.startIncommingCallActivity(this);
```
Comment (ianblenke) [inline src/org/linphone/LinphoneActivity.java]: Please change this from "Incomming" to "Incoming".

File: src/org/linphone/LinphoneActivity.java [hunk 10]
```diff
@@ -1519,9 +1523,11 @@ protected void onNewIntent(Intent intent) {
 				incoming.add(LinphoneCall.State.IncomingReceived);
 				if (LinphoneUtils.getCallsInState(getLc(), incoming).size() > 0) {
 					if (InCallActivity.isInstanciated()) {
-						InCallActivity.instance().startIncomingCallActivity();
+						LinphoneManager.startIncommingCallActivity(this);
+						//InCallActivity.instance().startIncomingCallActivity();
 					} else {
-						startActivity(new Intent(this, IncomingCallActivity.class));
+						LinphoneManager.startIncommingCallActivity(this);
```
Comment (ianblenke) [inline src/org/linphone/LinphoneActivity.java]: Please change these from "Incomming" to "Incoming".

File: src/org/linphone/LinphoneManager.java [hunk 11]
```diff
@@ -1679,4 +1679,26 @@ public void friendListRemoved(LinphoneCore lc, LinphoneFriendList list) {
                 // TODO Auto-generated method stub
 
         }
+
+
+	public static void startIncommingCallActivity(Context context )
+	{
+		LinphoneCore lc = getLc();
+		Log.d("startIncommingCallActivity new call");
```
Comment (ianblenke) [inline src/org/linphone/LinphoneManager.java]: Please change these from "Incomming" to "Incoming".

File: submodules/linphone [hunk 12]
```diff
@@ -1 +1 @@
-Subproject commit 73ee4c1f7f7eb2316e4ba81dd6642418e9f17757
+Subproject commit 44ae1322d2ec184d909c7b97167182d5fadbdffd
```
Comment (ianblenke) [inline submodules/linphone]: I'm going to want to know why we are updating the submodules here.

Comment (zackm0571) [inline submodules/linphone]: Agreed. Submodules should match the SHA1 of our master. Do not merge until this is fixed.

**Pull request merged**